api/app.rb

```diff
@@ -69,16 +70,16 @@ class App < Sinatra::Base
   helpers do
     attr_accessor :current_user
 
-    def url_from_port(current_user, port)
-      # Return a protocol relative URL to the backend server.
+    def build_payload(current_user, password, port, dir)
       mount_point = FlightFileManager.config.mount_point
-      "//#{request.host}:#{request.port}#{mount_point}/backend/#{current_user}"
-    end
-
-    def build_payload(current_user, password, port)
+      url_path = if dir
+                   File.join(mount_point, 'backend', current_user, 'fs', dir)
+                 else
+                   File.join(mount_point, 'backend', current_user)
```

Comment:
I'm pretty sure the majority of our applications won't do the right thing if ran on windows :rofl:

`File.join` will work correctly, but `dir` and `mount_point` may contain `/` which would break

Reply:
Yeah I'm not bothered about running on Windows.  It just seemed an odd way to join URL paths.  I'm sure its fine for our usage.

Reply:
Oh yeah, I use `File.join` for URL paths a bit. Mainly because `URI.join` has some odd behaviour when dealing with relative paths which trips me up.